src/models/event.ts

```diff
@@ -579,9 +579,18 @@ export class MatrixEvent extends TypedEventEmitter<MatrixEventEmittedEvents, Mat
         const relatesTo = this.getWireContent()?.["m.relates_to"];
         if (relatesTo?.rel_type === THREAD_RELATION_TYPE.name) {
             return relatesTo.event_id;
-        } else {
-            return this.getThread()?.id || this.threadId;
         }
+        if (this.thread) {
+            return this.thread.id;
+        }
+        if (this.threadId != undefined) {
```

Comment:
Why are we using `!=` over `!==` in this scenario?
If there's some special case, can we add a comment explaining it, and if not can we use `!==` instead?

Reply:
Fixes